common/scala/src/main/scala/org/apache/openwhisk/core/entity/Parameter.scala

```diff
@@ -80,15 +81,33 @@ protected[core] class Parameters protected[entity] (private val params: Map[Para
     params.keySet filter (params(_).init) map (_.name)
   }
 
+  protected[core] def getMap = {
+    params
+  }
   protected[core] def toJsArray = {
     JsArray(params map { p =>
-      if (p._2.init) {
-        JsObject("key" -> p._1.name.toJson, "value" -> p._2.value.toJson, "init" -> JsTrue)
-      } else JsObject("key" -> p._1.name.toJson, "value" -> p._2.value.toJson)
+      val init = p._2.init match {
+        case true => Some("init" -> p._2.init.toJson)
+        case _    => None
+      }
+      val encrypt = p._2.encryption match {
+        case (JsNull) => None
+        case _        => Some("encryption" -> p._2.encryption)
+      }
+      // Have do use this slightly strange construction to get the json object order identical.
+      JsObject(ListMap() ++ encrypt ++ init ++ Map("key" -> p._1.name.toJson, "value" -> p._2.value.toJson))
     } toSeq: _*)
   }
 
-  protected[core] def toJsObject = JsObject(params.map(p => (p._1.name -> p._2.value.toJson)))
+  protected[core] def toJsObject =
+    JsObject(params.map(p => {
+      val newValue =
+        if (p._2.encryption == JsNull)
+          p._2.value.toJson
+        else
```

Comment:
when encryption is not enabled on the parameter, `init` is omitted (this is the "old" format) but when encryption is enabled, the `init` field is present, why?

Reply:
OK, I understand why this is being done. I think it's unnecessary. Will attempt to remove it.